src/normalizer.ts

```diff
@@ -338,30 +339,27 @@ class V1SchemaNormalizer {
     if (!s || typeof s !== 'object' || Array.isArray(s)) return undefined
     if (s.xtpType) return s.xtpType
 
-    if ((s.type && s.type === 'object') ||
-      (s.properties && s.properties.length > 0)) {
-
+    if (s.properties && s.properties.length > 0) {
       s.type = 'object'
 
       const properties: XtpNormalizedType[] = []
-      if (s.properties) {
-        for (const pname in s.properties) {
-          const p = s.properties[pname]
-
-          const t = this.annotateType(p, [...path, 'properties', p.name ?? pname])
-          if (t) {
-            p.xtpType = t
-            properties.push(t)
-          }
+      for (const pname in s.properties) {
+        const p = s.properties[pname]
+
+        const t = this.annotateType(p, [...path, 'properties', p.name ?? pname])
+        if (t) {
+          p.xtpType = t
+          properties.push(t)
         }
-        return new ObjectType(s.name || '', properties, s)
-      } else {
-        // untyped object
-        return new ObjectType('', [], s)
       }
+      return new ObjectType(s.name || '', properties, s)
     }
 
     if (s.$ref) {
+      // don't recurse if the type is known
+      if (s.type) {
+        return undefined
+      }
```

Comment:
can you explain what is happening here?

Reply:
Before when `annotateType` was called on an item with `type: object` that also has `$ref`, it constructed `ObjectType` as if it was an untyped object. I think this was incorrect, but it allowed `invalid-cycle-doc` to pass. The new object detection code is only checking for typed objects, so it's solely dependent on `properties` and not `type`.

If I remove the check, `annotateType` recurses indefinitely. We could make the check more fine checking the type is `object`, but `type` being set on an item with a `$ref`, I think signals that an item has already been visited.